File: C#/BogoSort.cs
using System;

class BogoSort
{
    static Random rand = new Random();

    static bool IsSorted(int[] arr)
    {
        for (int i = 0; i < arr.Length - 1; i++)
        {
            if (arr[i] > arr[i + 1])
                return false;
        }
        return true;
    }

    static void Shuffle(int[] arr)
    {
        for (int i = 0; i < arr.Length; i++)
        {
            int j = rand.Next(arr.Length);
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }

    public static void Sort(int[] arr)
    {
        while (!IsSorted(arr))
        {
            Shuffle(arr);
        }
    }

    public static void Main()
    {
        int[] arr = { 3, 2, 5, 1, 4 };
        Sort(arr);
        Console.WriteLine(string.Join(" ", arr));
    }
}

File: Python/bogo_sort.py
import random

def is_sorted(arr):
    return all(arr[i] <= arr[i + 1] for i in range(len(arr) - 1))

def bogo_sort(arr):
    while not is_sorted(arr):
        random.shuffle(arr)
    return arr

arr = [3, 7, 4, 8, 6, 2, 0]
print(bogo_sort(arr))